xcube/util/zarrinsert.py

```diff
@@ -0,0 +1,126 @@
+        # --> files of variables, excluding "lat" and "lon" need to be renamed
+        _rename_file(src_path, src_idx, new_idx)
+        # Preparing the destination directory to be ready for single time stam to be merged
+        # --> files of variables, excluding "lat" and "lon" need to be renamed
+        # The renaming needs to happen in reversed order and starting at the index of nearest above value:
+        for i in reversed(range(new_idx, ds.time.shape[0])):
+            _rename_file(dst_path, i, (i + 1))
+        # Final step: copy the single time stamp files into the destination zarr and adjusting .zarray to the change.
+        _copy_into_target(src_path, dst_path, new_idx)
+        return True
+    else:
+        return False
+
+
+def _check_if_unique(src_path, dst_path):
+    """Check if to be added time stamp is unique """
+    ds = xr.open_zarr(dst_path)
+    ds_single = xr.open_zarr(src_path)
+    mask = np.equal(ds.time.values, ds_single.time.values)
+    ds.close()
+    ds_single.close()
+    if not mask.any():
+        return True
+    else:
+        return False
+
+
+def _rename_file(path_to_ds, old_index, new_time_i):
+    """Renaming files within the directories according to new time index."""
+    ds = xr.open_zarr(path_to_ds)
+    variables = ds.variables
+    ds.close()
+    for v in variables:
+        if (v != 'lat') and (v != 'lon') and (v != 'lat_bnds') and (v != 'lon_bnds'):
+            path = os.path.join(path_to_ds, v)
+            for root, dirs, files in os.walk(path):
+                for filename in files:
+                    parts1 = filename.split('.', 1)[0]
+                    if parts1 == (str(old_index)) and (v != "time"):
+                        parts2 = filename.split('.', 1)[1]
+                        new_name = (str(new_time_i) + '.{}').format(parts2)
+                        os.rename(os.path.join(path, filename), os.path.join(path, new_name))
+                    elif parts1 == (str(old_index)) and (v == "time"):
+                        os.rename(os.path.join(path, filename), os.path.join(path, str(new_time_i)))
+
+
+def _copy_into_target(src_path, dst_path, src_index):
+    """Copy the files with the new time stamp into the existing zarr directory."""
+    ds = xr.open_zarr(src_path)
+    variables = ds.variables
+    ds.close()
+    for v in variables:
+        if (v != 'lat') and (v != 'lon') and (v != 'lat_bnds') and (v != 'lon_bnds'):
+            path = os.path.join(src_path, v)
+            for root, dirs, files in os.walk(path):
+                for filename in files:
+                    parts1 = filename.split('.', 1)[0]
+                    if parts1 == str(src_index):
+                        shutil.copyfile((os.path.join(src_path, v, filename)), (os.path.join(dst_path, v, filename)))
+            _adjust_zarray(dst_path, v)
+
+
+def _adjust_zarray(dst_path, variable):
+    """Changing the shape for time in the .zarray file."""
```

Comment:
The function does not look very "time"-specific.

The name `_adjust_zarray` does not tell what it actually does. Use better name e.g. `_expand_1d_var_shape`